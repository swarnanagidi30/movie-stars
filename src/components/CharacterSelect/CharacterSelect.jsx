import React from 'react';
import TextField from '@material-ui/core/TextField';
import Autocomplete from '@material-ui/lab/Autocomplete';
import CircularProgress from '@material-ui/core/CircularProgress';
import fetch from 'node-fetch';

export default function CharacterSelect({ onChange }) {
    const [open, setOpen] = React.useState(false);
    const [options, setOptions] = React.useState([]);
    const loading = open && options.length === 0;

    React.useEffect(() => {
        let active = true;

        if (!loading) {
            return undefined;
        }

        (async () => {
            const response = await fetch('https://swapi.dev/api/people/');
            const peoples = await response.json();
            if (active) {
                setOptions(peoples.results);
            }
        })();

        return () => {
            active = false;
        };
    }, [loading]);

    React.useEffect(() => {
        if (!open) {
            setOptions([]);
        }
    }, [open]);

    return (
        <Autocomplete
            open={open}
            onChange={(event, newValue) => {
                if (onChange) {
                    onChange(newValue);
                }
                // setValue(newValue)
            }}
            onOpen={() => {
                setOpen(true);
            }}
            onClose={() => {
                setOpen(false);
            }}
            getOptionSelected={(option, value) => option.name === value.name}
            getOptionLabel={(option) => option ? option.name : ''}
            options={options}
            loading={loading}
            renderInput={(params) => (
                <TextField
                    {...params}
                    label="Character"
                    variant="outlined"
                    InputProps={{
                        ...params.InputProps,
                        endAdornment: (
                            <React.Fragment>
                                {loading ? <CircularProgress color="inherit" size={20} /> : null}
                                {params.InputProps.endAdornment}
                            </React.Fragment>
                        ),
                    }}
                />
            )}
        />
    );
}