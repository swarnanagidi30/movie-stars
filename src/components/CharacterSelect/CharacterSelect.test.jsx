import React from 'react';
jest.mock('node-fetch');

// import fetch, { Response } from 'node-fetch';

import { createShallow } from '@material-ui/core/test-utils';
import { ThemeProvider } from '@material-ui/core/styles';
import CharacterSelect from './CharacterSelect';
import toJson from "enzyme-to-json"

describe("<CharacterSelect />", () => {
    const renderShallowUntilComponent = createShallow({ "untilSelector": "CharacterSelect" })
    let Component
    beforeEach(() => {

        Component = renderShallowUntilComponent(<ThemeProvider theme={{ success: { main: '#fff' } }}><CharacterSelect /></ThemeProvider>)
    })

    it("loads", () => {
        expect(typeof Component).toBe("object")
    })

    it("has a <Autocomplete> container", () => {
        expect(Component.find("WithStyles(ForwardRef(Autocomplete))").length).toBeGreaterThanOrEqual(1)
    })

    it("snapshots", () => {
        expect(toJson(Component)).toMatchSnapshot()
    })
})
