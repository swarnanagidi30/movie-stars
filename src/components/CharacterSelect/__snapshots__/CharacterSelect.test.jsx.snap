// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterSelect /> snapshots 1`] = `
<WithStyles(ForwardRef(Autocomplete))
  getOptionLabel={[Function]}
  getOptionSelected={[Function]}
  loading={false}
  onChange={[Function]}
  onClose={[Function]}
  onOpen={[Function]}
  open={false}
  options={Array []}
  renderInput={[Function]}
/>
`;
